src/imu_diag_ros_publisher.cpp

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************//**
+*   @file   imu_diag_ros_publisher.cpp
+*   @brief  Implementation for imu diag publisher
+*   @author Vasile Holonec (Vasile.Holonec@analog.com)
+********************************************************************************
+* Copyright 2023(c) Analog Devices, Inc.
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "imu_ros2/imu_diag_ros_publisher.h"
+#include <thread>
+
+
+ImuDiagRosPublisher::ImuDiagRosPublisher(std::shared_ptr<rclcpp::Node>& node)
+{
+    init(node);
+}
+
+ImuDiagRosPublisher::~ImuDiagRosPublisher()
+{
+    delete m_dataProvider;
+}
+
+void ImuDiagRosPublisher::init(std::shared_ptr<rclcpp::Node> &node)
+{
+    m_node = node;
+    m_publisher = node->create_publisher<imu_ros2::msg::ImuDiagData>("imudiagdata", 10);
+}
+
+void ImuDiagRosPublisher::setMessageProvider(ImuDiagDataProviderInterface *dataProvider)
+{
+    m_dataProvider = dataProvider;
+}
+
+void ImuDiagRosPublisher::run()
+{
+    std::thread::id this_id = std::this_thread::get_id();
+    std::cout << "thread " << this_id << " started...\n";
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "startThread: '%d'", this_id);
+
+    rclcpp::WallRate loopRate(0.01);
+
+    int count = 0;
+    while (rclcpp::ok())
+    {
+
+      //std::thread::id this_id = std::this_thread::get_id();
+       // std::cout << "thread " << this_id << " running...\n";
+       // RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
+
+        //auto started = std::chrono::high_resolution_clock::now();
```

Comment:
In general, I do not think commented code lines are a good idea when pushing to main branch.
Let's try and remove any commented code lines if they are no longer needed.